tests/backup/backup_resiliency_test.go

```diff
@@ -758,3 +761,240 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
+			log.InfoD("Restoring the backups taken")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting admin context")
+			for _, backupName := range backupNames {
+				sem <- struct{}{}
+				restoreName := fmt.Sprintf("%s-restore", backupName)
+				restoreNames = append(restoreNames, restoreName)
+				wg.Add(1)
+				go func(backupName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					defer func() { <-sem }()
+					_, err = CreateRestoreWithoutCheck(restoreName, backupName, nil, destinationClusterName, orgID, ctx)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
+				}(backupName)
+			}
+			wg.Wait()
+			log.Infof("The list of restores are: %v", restoreNames)
+		})
+		Step("Scaling MongoDB statefulset replica to 0 and back to original replica while restore is in progress", func() {
+			log.InfoD("Scaling MongoDB statefulset replica to 0 while restore is in progress")
+			*statefulSet.Spec.Replicas = scaledDownReplica
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling to 0 is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling down MongoDB statefulset replica to 0"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == scaledDownReplica, true, "Getting mongodb statefulset replica after scaling down")
+			log.InfoD("Sleeping for 1 minute so that at least one request is hit to mongodb for the restores taken")
+			time.Sleep(1 * time.Minute)
+			log.InfoD("Scaling MongoDB statefulset to original replica while restore is in progress")
+			statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNS)
+			*statefulSet.Spec.Replicas = originalReplicaCount
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling back to original replica is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling MongoDB statefulset back to original replica"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == originalReplicaCount, true, "Verify mongodb statefulset replica after scaling back to original")
+			log.Infof("Verify that at least one mongodb pod is in Ready state")
+			mongoDBPodStatus := func() (interface{}, bool, error) {
+				statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNS)
+				if err != nil {
+					return "", true, err
+				}
+				if statefulSet.Status.ReadyReplicas < 1 {
+					return "", true, fmt.Errorf("no mongodb pods are ready yet")
+				}
+				return "", false, nil
+			}
+			_, err = task.DoRetryWithTimeout(mongoDBPodStatus, 5*time.Minute, 30*time.Second)
+			log.Infof("Number of mongodb pods in Ready state are %v", statefulSet.Status.ReadyReplicas)
+			dash.VerifyFatal(statefulSet.Status.ReadyReplicas > 0, true, "Verifying that at least one mongodb pod is in Ready state")
```

Comment:
Same here

Reply:
I tried it manually and observed that immediately after 1 pod is ready backup goes to success state